Realtime-josh/EPIC_Mail
# Ft user login v2 164687824

**What Does This PR Do?**
This task is a feature. User can login to account only after user must have signed up with validated data.

**Description of task to be Completed**
This login feature allows registered users to login to the site to access all of the features that the account gives privilege. User should be able to provide an email and password parameter in the request body fields which should be verified that it exists in the database. Only specific emails with @epicmail.com are accepted. passwords of character lengths less than 6 are invalid. 
After email and password fields are being validated, A database call is initiated. It should make certain that the email exists for signed up user. Encrypted passwords are decrypted using bcrypt.compare in order to check for a match result gotten from body requests and database calls. If all test cases passes, a payload is built upon user data gotten from the database.
A token is being generated using jsonwebtoken. Authorization headers are now set.

**How should this be manually tested**
clone git hub repository https://github.com/Realtime-josh/EPIC_Mail on branch ft-user-login-v2-164687824.
-Hit npm install to install package files
-Hit npm start to run files on your local machine
-Ensure that a local database is set up on machine(postgresql)
-On postman route can be accessed using http://localhost:{preferred port number}/api/v2/auth/login
-This is a POST request 
[Finishes ft-user-login-v2-164687824]
